### PR TITLE
Update _reformat_low to not run kwarg dicts through parse_input

### PR DESCRIPTION
**Pull request by whiteinge**
### What does this PR do?

Keyword args were incorrectly getting run through `parse_input`. E.g., `{'fun': 'jobs.list_jobs', 'jid': '1234'}` would run the `jid` kwarg through that function and then YAML would try to parse the argument as a number.

load_args_and_kwargs() call already calls parse_input() which already
calls condition_input() so I think we can do with just the one call.

This also removes a duplicate `{..., 'kwarg': {'kwarg': {...}}}` in the output.

### What issues does this PR fix or reference?

#38962

### New Behavior

I'm 43% sure this doesn't regress anything...

### Tests written?

No

@DmitryKuzmenko you've done some work here, do you mind giving this a look?

### CODE REVIEW

**Comment by s0undt3ch** [comment]
43%! Not 43.7%?

**Comment by thatch45** [comment]
This honestly gives me the willies, we rely quite heavily on the parse_input stuff. I would almost manually search for a jid field and make sure it is a string.

**Comment by cachedout** [comment]
This feels like a good direction to me but I am resistant to it being in 2016.11. IMHO, develop is the place for this.

**Comment by whiteinge** [comment]
We're still calling `parse_input` because `salt.minion.load_args_and_kwargs` calls it. Before this change we were calling it against a `{"jid": "1234"}` dict instead of only a `jid=1234` string.

That said, I don't actually know how to exercise this code since the `low` dict cannot contain `name=value` string pairs. I can't tell why it was added in the first place.

But I do know the CLI does not go through this code path:

```
% salt-run test.arg foo bar=true baz='{qux: Qux}' --out=json
{
    "args": [
        "foo"
    ],
    "kwargs": {
        "baz": {
            "qux": "Qux"
        },
        "bar": true
    }
}
```

I'm also pretty sure the Reactor and Orchestrate both call the `RunnerClient.cmd` method and not this `cmd_sync` method.

**Comment by DmitryKuzmenko** [comment]
@whiteinge I just agree with Mike. We have to test it well to be 146%©️ sure.

**Comment by whiteinge** [comment]
I misspoke in my comment in the associated issue: This change does not affect Runner calls via CLI/Reactor/Orchestrate; this only affects calls via salt-api or Python API calls in user scripts (which, as mentioned, cannot send positional args and thus cannot benefit from `parse_input`).

Since this is a bugfix and not a feature addition, I'll do a deeper dive for why this was added in the first place to try and raise my confidence level.

**Comment by thatch45** [comment]
Thanks @whiteinge that all makes me feel a lot better about this change

**Comment by whiteinge** [comment]
I was able to track down the history on this and everything is now clear. Full explanation in the commit messages above. tl;dr: The commit that introduced the bug was trying to _add_ `parse_input` support for positional args, but added it for keyword args instead. And I signed off on it back then 😳 . In fairness to the author, this part of the code has a long history and is a tad hard to follow.

This PR is now updated to fix the bug and fulfil the original intention.

As penance for misreading the original PR I have added a test for the arg/kwarg parsing behavior.

**Comment by whiteinge** [comment]
Oh, this also re-enables some RunnerClient tests that had been marked for skipping. I'm not clear on why. They run locally so hopefully they'll run on Jenkins too.

**Comment by whiteinge** [comment]
TIL that salt-run takes an eauth flag.

The failing integration test actually tries to create a PAM user on my local system (O_o) so I'll revisit that on Monday when I have a little time.

**Comment by cachedout** [comment]
Go Go Jenkins!

**Comment by cachedout** [comment]
Go Go Jenkins!

**Comment by cachedout** [comment]
How do we all feel about which branch this should go into?

**Comment by whiteinge** [comment]
I'm not unbiased so I'll say my piece then defer to the group's opinion:

After diving into the history, I'm pretty confident with where this ended up. No new functionality, only fixes. The bug mentioned in OP _could_ have a less intrusive change that bypasses the bug without fixing it but we'll need to fix it eventually.
